src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -16,18 +19,23 @@
     protected static final String SORTBY = "sortBy";
     protected static final String ORDER = "order";
     private ProductDao productDao;
+    private RecentViewService recentViewService;
 
     @Override
     public void init() throws ServletException {
         productDao = ArrayListProductDao.getInstance();
+        recentViewService = HttpSessionRecentViewService.getInstance();
     }
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
         String query = request.getParameter(QUERY);
-        SortBy sortBy = request.getParameter(SORTBY) != null ? SortBy.valueOf(request.getParameter(SORTBY)) : null;
-        SortingOrder order = request.getParameter(ORDER) != null ? SortingOrder.valueOf(request.getParameter(ORDER)) : SortingOrder.ASC;
-
+        SortBy sortBy = request.getParameter(SORTBY) != null ? SortBy.valueOf(request.getParameter(SORTBY).toUpperCase())
+                : null;
+        SortingOrder order = request.getParameter(ORDER) != null ? SortingOrder.valueOf(request.getParameter(ORDER).toUpperCase())
+                : SortingOrder.ASC;
+        RecentView recentView = recentViewService.getRecentView(request);
+        request.setAttribute("recentView", recentView.getRecentlyViewed());
         if (query == null) {
```

Comment:
Why not to use the same logic as with SortingOrder and SortBy?
query = request.getParameter(QUERY) != null ? request.getParameter(QUERY) : " "